ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -1317,72 +1120,36 @@ void VKContext::DrawIndexed(int vertexCount, int offset) {
 	size_t vbBindOffset = push_->Push(vbuf->GetData(), vbuf->GetSize(), &vulkanVbuf);
 	size_t ibBindOffset = push_->Push(ibuf->GetData(), ibuf->GetSize(), &vulkanIbuf);
 
-	vkCmdBindPipeline(cmd_, VK_PIPELINE_BIND_POINT_GRAPHICS, curPipeline_->vkpipeline);
-
 	VkDescriptorSet descSet = GetOrCreateDescriptorSet(vulkanUBObuf);
-	vkCmdBindDescriptorSets(cmd_, VK_PIPELINE_BIND_POINT_GRAPHICS, pipelineLayout_, 0, 1, &descSet, 1, &ubo_offset);
 
-	VkBuffer buffers[1] = { vulkanVbuf };
-	VkDeviceSize offsets[1] = { vbBindOffset };
-	vkCmdBindVertexBuffers(cmd_, 0, 1, buffers, offsets);
-
-	vkCmdBindIndexBuffer(cmd_, vulkanIbuf, ibBindOffset, VK_INDEX_TYPE_UINT32);
-	vkCmdDrawIndexed(cmd_, vertexCount, 1, 0, offset, 0);
+	renderManager_.BindPipeline(curPipeline_->vkpipeline);
+	// TODO: blend constants, stencil, viewports should be here, after bindpipeline..
+	renderManager_.DrawIndexed(pipelineLayout_, descSet, 1, &ubo_offset, vulkanVbuf, (int)vbBindOffset, vulkanIbuf, (int)ibBindOffset, vertexCount, 1, VK_INDEX_TYPE_UINT32);
 }
 
 void VKContext::DrawUP(const void *vdata, int vertexCount) {
-	ApplyDynamicState();
-
 	VkBuffer vulkanVbuf, vulkanUBObuf;
 	size_t vbBindOffset = push_->Push(vdata, vertexCount * curPipeline_->stride[0], &vulkanVbuf);
 	uint32_t ubo_offset = (uint32_t)curPipeline_->PushUBO(push_, vulkan_, &vulkanUBObuf);
 
-	vkCmdBindPipeline(cmd_, VK_PIPELINE_BIND_POINT_GRAPHICS, curPipeline_->vkpipeline);
-
-	VkBuffer buffers[1] = { vulkanVbuf };
-	VkDeviceSize offsets[1] = { vbBindOffset };
-	vkCmdBindVertexBuffers(cmd_, 0, 1, buffers, offsets);
-
 	VkDescriptorSet descSet = GetOrCreateDescriptorSet(vulkanUBObuf);
-	vkCmdBindDescriptorSets(cmd_, VK_PIPELINE_BIND_POINT_GRAPHICS, pipelineLayout_, 0, 1, &descSet, 1, &ubo_offset);
-	vkCmdDraw(cmd_, vertexCount, 1, 0, 0);
+
+	renderManager_.BindPipeline(curPipeline_->vkpipeline);
+	// TODO: blend constants, stencil, viewports should be here, after bindpipeline..
+	renderManager_.Draw(pipelineLayout_, descSet, 1, &ubo_offset, vulkanVbuf, (int)vbBindOffset, vertexCount);
 }
 
 // TODO: We should avoid this function as much as possible, instead use renderpass on-load clearing.
```

Comment:
In *theory* this is no longer true and it's optimized even if we use this func, right?  I mean, we should still do this but it's no longer a TODO anyway...

-[Unknown]

Reply:
Right, removing comment.